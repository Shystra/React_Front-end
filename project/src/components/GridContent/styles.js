import styled, { css } from 'styled-components';


export const GridContent = styled.div`
    ${({ theme }) => css`
    
    
    `}
`;