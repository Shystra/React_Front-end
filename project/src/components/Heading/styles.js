import styled, { css } from 'styled-components';



const titleSize = {
  small: (theme) => css`
    font-size: ${theme.font.sizes.medium};`,

  medium: (theme) => css`
    font-size: ${theme.font.sizes.large};`,

  

  huge: (theme) => css`
    font-size: ${theme.font.sizes.huge};
    
  `,

  big: (theme) => css`
    font-size: ${theme.font.sizes.xlarge};
    ${mediaFont(theme)};`,
};

const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.huge};
    background: purple;
    
  }
`;


const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;



export const Title = styled.h1`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.primaryColor : theme.colors.white};
    ${titleSize[size](theme)};
    ${titleCase(uppercase)};
  `}
`;