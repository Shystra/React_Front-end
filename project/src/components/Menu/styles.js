import styled, {css} from 'styled-components';

export const Container = styled.div`
    ${({ theme }) => css`
        position: fixed;
        z-index: 5;
        top: 0;
        left: 0;
        right: 0;
        bottom: 0;
        border-bottom: ${theme.colors.mediumGray}; 
    
    `}
`;

export const MenuContainer = styled.div`
    ${({ theme }) => css`
        display: flex;
        justify-content: space-between;
        align-items: center;

        @media ${theme.media.lteMedium} {
            display: block;
            text-align: center;
            /* padding: ${theme.spacings.xxlarge}; */
        }
    
    `}
`;
export const Button = styled.button`
    ${({ theme }) => css`
        
    
    `}
`;
