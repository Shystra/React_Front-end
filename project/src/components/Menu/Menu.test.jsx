import { screen } from '@testing-library/react';
import { Menu } from '.'
import { renderTheme } from '../styles/render_theme';
import linksMock from '../NavLinks/mock';

const logoData = { 
    text: 'Logo',
    link: '#target',
};


describe('<Menu />', () => {
    it('should render Logo and Main Menu Nav', () => {
        const {contianer} = renderTheme(<Menu links={linksMock} logoData={logoData}/>);

        expect(screen.getByRole('heading', { name: 'Logo' })).toBeInTheDocument();
        expect(screen.getByRole('navigation', { name: 'Main menu' })).toBeInTheDocument();

        expect (contianer).toMatchSnapshot();


    });
});