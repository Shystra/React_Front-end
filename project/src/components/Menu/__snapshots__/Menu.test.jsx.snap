// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should render Logo and Main Menu Nav 1`] = `undefined`;
